creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -498,16 +499,19 @@ class SnapbackSM {
     // TODO change to chunked parallel
     await Promise.all(userReplicaSets.map(async (user) => {
       try {
-        const { wallet, endpoint: secondary } = user
+        const { wallet, primary, secondary1, secondary2, endpoint: secondary } = user
 
-        // TODO - throw on null wallet (is this needed?)
+        // Short-circuit if primary is not self - this function is meant to be called from primary to secondaries only
+        if (primary !== this.endpoint) {
+          this.logError(`issueSyncRequests || Can only be called by user's primary. User ${wallet} - replicaset [${primary}, ${secondary1}, ${secondary2}].`)
+          return
+        }
 
-        // Determine if secondary requires a sync by comparing clock values against primary (this node)
+        // Determine if secondary requires a sync by comparing clock values against primary (self)
         const userPrimaryClockVal = userPrimaryClockValues[wallet]
         const userSecondaryClockVal = secondaryNodesToUserClockStatusesMap[secondary][wallet]
-        const syncRequired = !userSecondaryClockVal || (userPrimaryClockVal > userSecondaryClockVal)
 
-        if (syncRequired) {
+        if (userPrimaryClockVal > userSecondaryClockVal) {
```

Comment:
nice

Reply:
maybe we should also do an explicit check here that both of these values are numbers prior to this comparison

Reply:
i think its prob fine here, if we start seeing issues can handle